server/grpc_service.go

```diff
@@ -125,18 +122,160 @@ func (s *GrpcServer) GetClusterInfo(ctx context.Context, _ *pdpb.GetClusterInfoR
 	}, nil
 }
 
+// GetMinTS implements gRPC PDServer. In PD service mode, it simply returns a timestamp.
+// In API service mode, it queries all tso servers and gets the minimum timestamp across
+// all keyspace groups.
+func (s *GrpcServer) GetMinTS(
+	ctx context.Context, request *pdpb.GetMinTSRequest,
+) (*pdpb.GetMinTSResponse, error) {
+	if err := s.validateRequest(request.GetHeader()); err != nil {
+		return &pdpb.GetMinTSResponse{
+			Header: s.wrapErrorToHeader(pdpb.ErrorType_UNKNOWN, err.Error()),
+		}, nil
+	}
+
+	var (
+		minTS *pdpb.Timestamp
+		err   error
+	)
+	if s.IsAPIServiceMode() {
```

Comment:
Should we check if the tso server has implemented this method?

Reply:
no need, right? Two reasons:
1. The tso service GetMinTS kvproto change and server side implementation is earlier than this pr, so this is only about the startup order of the API servers and tso servers during service upgrade.
2. GetMinTS will always fail during service upgrade and there is no correctness related issue because of the validation during aggregation. Because GetMinTS is an operational request with low QPS, failure during server upgrade should be acceptable.